app/src/org/commcare/android/tasks/TemplatePrinterTask.java

```diff
@@ -0,0 +1,253 @@
+package org.commcare.android.tasks;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.commcare.android.util.TemplatePrinterUtils;
+
+
+/**
+ * Asynchronous task for populating an html document with data.
+ *
+ * @author Richard Lu
+ * @author amstone
+ */
+public class TemplatePrinterTask extends AsyncTask<Void, Void, Integer> {
+
+    /**
+     * The 4 result codes that can be sent back by this task
+     */
+    public static final int SUCCESS = 0;
+    public static final int IO_ERROR = 1;
+    public static final int VALIDATION_ERROR_MUSTACHE = 2;
+    public static final int VALIDATION_ERROR_CHEVRON = 3;
```

Comment:
Would be sweet if these were an enum. It would also _slightly_ clean up the TemplatePrinterActivity.onFinished switch statement